storage/src/main/java/com/tencent/rss/storage/handler/impl/HdfsClientReadHandler.java

```diff
@@ -63,96 +58,88 @@ public HdfsClientReadHandler(
     this.hadoopConf = hadoopConf;
+    this.readHandlerIndex = 0;
     String fullShufflePath = ShuffleStorageUtils.getFullShuffleDataFolder(storageBasePath,
         ShuffleStorageUtils.getShuffleDataPathWithRange(appId,
             shuffleId, partitionId, partitionNumPerRange, partitionNum));
     init(fullShufflePath);
-    readerIterator = indexReaderMap
-        .entrySet()
-        .stream()
-        .sorted(Entry.comparingByKey())
-        .collect(Collectors.toList())
-        .iterator();
   }
 
-  public ShuffleIndexResult readShuffleIndex() {
-    long start = System.currentTimeMillis();
+  protected void init(String fullShufflePath) {
+    FileSystem fs;
+    Path baseFolder = new Path(fullShufflePath);
     try {
-      byte[] indexData = currentIndexReader.readIndex();
-      LOG.info("Read index files for shuffleId[{}], partitionId[{}] for {} ms",
-          shuffleId, partitionId, System.currentTimeMillis() - start);
-      return new ShuffleIndexResult(indexData);
-    } catch (Exception e) {
-      LOG.info("Fail to read index files for shuffleId[{}], partitionId[{}]", shuffleId, partitionId);
+      fs = ShuffleStorageUtils.getFileSystemForPath(baseFolder, hadoopConf);
+    } catch (IOException ioe) {
+      throw new RuntimeException("Can't get FileSystem for " + baseFolder);
     }
 
-    return new ShuffleIndexResult();
-  }
+    FileStatus[] indexFiles;
+    String failedGetIndexFileMsg = "Can't list index file in  " + baseFolder;
 
-  public ShuffleDataResult readShuffleData(ShuffleDataSegment shuffleDataSegment) {
-    int expectedLength = shuffleDataSegment.getLength();
-    if (expectedLength < 0) {
-      LOG.error("Invalid segment {}", shuffleDataSegment);
-      return null;
+    try {
+      // get all index files
+      indexFiles = fs.listStatus(baseFolder,
+          file -> file.getName().endsWith(Constants.SHUFFLE_INDEX_FILE_SUFFIX));
+    } catch (Exception e) {
+      LOG.error(failedGetIndexFileMsg, e);
+      return;
     }
 
-    byte[] data = currentDataReader.readData(shuffleDataSegment.getOffset(), expectedLength);
-    if (data.length != expectedLength) {
-      LOG.error("Fail to read expected[{}] data, actual[{}] and segment is {}",
-          expectedLength, data.length, shuffleDataSegment);
-      return null;
+    if (indexFiles != null) {
+      for (FileStatus status : indexFiles) {
+        LOG.info("Find index file for shuffleId[" + shuffleId + "], partitionId["
+            + partitionId + "] " + status.getPath());
+        String fileNamePrefix = getFileNamePrefix(status.getPath().getName());
+        try {
+          HdfsFileReader indexReader = createHdfsReader(
+              fullShufflePath, ShuffleStorageUtils.generateIndexFileName(fileNamePrefix), hadoopConf);
+          HdfsFileReader dataReader = createHdfsReader(
+              fullShufflePath, ShuffleStorageUtils.generateDataFileName(fileNamePrefix), hadoopConf);
+          HdfsShuffleReadHandler hdfsShuffleFileReader =
```

Comment:
indexReader and dataReader should be included in HdfsShuffleReadHandler

Reply:
OK